react/LocationInput.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState, useEffect } from 'react'
+import { useAddressContext } from 'vtex.address-context/AddressContext'
+import { IconSearch, Input, ButtonPlain } from 'vtex.styleguide'
+import { FormattedMessage } from 'react-intl'
+import { useLazyQuery } from 'react-apollo'
+
+import GET_ADDRESS_FROM_POSTAL_CODE from './graphql/getAddressFromPostalCode.graphql'
+
+interface Props {
+  onSuccess?: (address: any) => void
+}
+
+const LocationInput: React.FC<Props> = ({ onSuccess }) => {
+  const { address, setAddress } = useAddressContext()
+  const [inputValue, setInputValue] = useState('')
+  const [getAddressFromPostalCode, { error, data, loading }] = useLazyQuery(
+    GET_ADDRESS_FROM_POSTAL_CODE
+  )
+
+  if (!address.country) {
+    throw new Error(
+      'The LocationField (Input) should be used when the country field is already filled'
+    )
+  }
+
+  useEffect(() => {
+    if (data) {
+      setAddress(data.getAddressFromPostalCode)
+      onSuccess?.(data.getAddressFromPostalCode)
+    }
+
+    if (error) {
```

Comment:
Why did you choose to not use the `if else` here?

Reply:
no particular reason, I could change if you'd like

Reply:
I was just curious because on `DeviceCoordinates.tsx` you (maybe Maxi, I don't know) have changed to use the `if else` and haven't done it here.
 ⛏ Would be nice to keep it consistent. I prefer this version, without the `else` but it's your call.

Reply:
okay! I altered it to an `else if` but I also prefer the two independent `if`s 😅